lib/smart_answer/resolve_state.rb

```diff
@@ -0,0 +1,80 @@
+module SmartAnswer
+  class ResolveState
```

Comment:
Can we better define this class as an entity (a noun) inline with OOP? I.e. StateResolver etc (or possibly these are just class methods on State?)

Reply:
Yeah I was thinking of this as a service class which are typically named with a verb but it's not really one of those so that seems reasonable. I think it benefits from being abstracted as a separate class given it's responsibility is quite different to State.